### DISCUSSION TITLE
PWA optimisation for IOS/IPadOS

### DISCUSSION THREAD

**Discussion by ShamanicArts** [noisedeck/noisedeck-docs#30] [Ideas]
Would be amazing to see a more reliable implementation of Noisedeck/Noisedeck Pro as Progressive Web App , currently NoisedeckPro on IPadOS is touch and go - safari it hardly ever initiates the graphics engine, but when it does it works well and can save. On chrome based browsers the engine almost always initiates but the save function is a lie and it doesn’t download

**Comment by jesshewitt** [comment]
Thanks for the report. I'll take a look at this...it's been a while since I've tested the PWA functionality on iPad.

**Reply by ShamanicArts** [reply]
![2E3FEBFF-606A-4D4C-8217-BC7F5B2048DB](https://user-images.githubusercontent.com/90007349/132254330-f6274854-a64e-4025-bbf2-fc4aba313d87.png)

facing some issues with noisedeck pro on IPadOS currently - I get this error message after logging in .

**Reply by jesshewitt** [reply]
I just tested with our iPad and it worked okay...have you tried reloading? Maybe it was a temporary networking issue.

**Reply by ShamanicArts** [reply]
I tried again just now and was fine, only the downloaded web app version had the issue - must have been some temporary network issue  however I have had this issue for 2 days on all browsers on iPad before reporting .

**Reply by jesshewitt** [reply]
I see...in the future, please do let us know right away if you're experiencing some sort of error that prevents you from using the application!
